packages/ckeditor5-link/src/linkui.js

```diff
@@ -154,7 +155,21 @@ export default class LinkUI extends Plugin {
 
 		// Execute link command after clicking the "Save" button.
 		this.listenTo( formView, 'submit', () => {
-			editor.execute( 'link', formView.urlInputView.fieldView.element.value, formView.getDecoratorSwitchesState() );
+			const { value } = formView.urlInputView.fieldView.element;
+
+			// It's good to check the config for the second time on submit.
+			// This will cover more dynamic cases, when protocol might change after the element has been rendered.
+			const defaultProtocol = editor.config.get( 'link.defaultProtocol' );
```

Comment:
We do not have dynamic config in CKEditor5 - #7383.

At this stage I'd made that static - changing manual test would be a PITA but this might be seen as a side-effect in the current code base. Other features read config once - at init stage.

Reply:
ps.: If dynamic options were already established somwhere else we can leave it but please add a comment in #7383.

Reply:
It will require changing the tests but only for the part when configured protocol is set would require a new describe block probably.